xrootdlib/structs/XrdXrootdMon/plugin.py

```diff
@@ -0,0 +1,74 @@
+import json
+import enum
+from typing import Union
+
+from ...utility import slot_repr
+
+
+class pluginType(str, enum.Enum):
+    """
+    Assumes binary header.
+    """
+    # Cache Context Manager 
+    isCCM = 'M'
+
+    # Proxy File Cache 
+    isPFC = 'C'
+
+    # TCP Connection Monitor 
+    isTCM = 'T'
+
+class PluginHdr(object):
+    """
+    Here tBeg, tEnd, sID should go
+    """
+    def __init__(self) -> None:
+        super().__init__()
+        raise NotImplementedError
```

Comment:
Kudos for being correct, but just raising is fine if the object is not functional anyway.
```suggestion
        raise NotImplementedError
```
Same for the other unimplemented types.